components/browser/icons/src/main/java/mozilla/components/browser/icons/loader/HttpIconLoader.kt

```diff
@@ -41,7 +41,7 @@ class HttpIconLoader(
         // recently: https://github.com/mozilla-mobile/android-components/issues/2591
 
         val downloadRequest = Request(
-            url = resource.url,
+            url = resource.url.trim(),
```

Comment:
We can go up higher:
Here we retrieve the values from the webextensions and I think that is the place where we may want to sanitize etc.:
https://github.com/mozilla-mobile/android-components/blob/master/components/browser/icons/src/main/java/mozilla/components/browser/icons/extension/IconMessage.kt#L82-L96

I think we should **not** do it in any of the `Client` implementations and let them continue fail if we pass invalid URLs to them.

Reply:
👍